CMSgov/design-system
# [WNMGDS-811] Tooltip updates

## Summary
- Update tooltip styling
   - Replace`display: flex` with `inline` so that tooltips can be used inline inside a sentence
   - Inherit font weight and size styles, so tooltips can easily be added inline inside a sentence
   - Add `vertical-align: middle` to vertically center tooltip icons instead of using flex styles
- Remove the `interactive` prop in favor of enabling it's functionality by default in order to satisfy new WCAG 2.2 requirement ensuring popovers are hoverable.
- Add `onClose` and `onOpen` props so users have access to those event handlers 
- Update and improve event handlers to improve focus/blur/click/touch behaviors across tooltips and tooltip dialogs. Also ensures that `onClose` and `onOpen` handlers are only called when relevant

## How to test
- Test new tooltip styles, `inline`, inherited font styles, and vertical alignment
- Test tooltips via keyboard
   - Tab through tooltips to ensure  focus and blur functionality is working correctly
   - Ensure the `esc` key allows tooltips to be closed
- Test tooltips via mouse and touch
   - Ensure tooltips can be opened and closed via clicking and touching the tooltip trigger
   - Ensure tooltip dialogs can be closed by clicking outside of the tooltip 
- Test tooltips with screen readers and on multiple browsers
- `http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-811/tooltip-updates`
